ddl/placement_policy_test.go

```diff
@@ -375,3 +375,221 @@ func (s *testDBSuite6) TestCreateTableWithPlacementPolicy(c *C) {
 	tk.MustExec("drop table if exists t")
 	tk.MustExec("drop placement policy if exists x")
 }
+
+func (s *testDBSuite6) TestCreateDatabaseWithPlacementPolicy(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	defer func() {
+		tk.MustExec("drop database if exists placementDb1")
+		tk.MustExec("drop placement policy if exists x")
+	}()
+
+	tk.MustExec("drop database if exists placementDb1")
+	// Direct placement option: special constraints may be incompatible with common constraint.
+	_, err := tk.Exec("create database placementDb1 " +
+		"PRIMARY_REGION=\"cn-east-1\" " +
+		"REGIONS=\"cn-east-1, cn-east-2\" " +
+		"FOLLOWERS=2 " +
+		"FOLLOWER_CONSTRAINTS=\"[+zone=cn-east-1]\" " +
+		"CONSTRAINTS=\"[+disk=ssd,-zone=cn-east-1]\"")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "conflicting label constraints: '-zone=cn-east-1' and '+zone=cn-east-1'")
+
+	tk.MustExec("create database placementDb1 " +
+		"PRIMARY_REGION=\"cn-east-1\" " +
+		"REGIONS=\"cn-east-1, cn-east-2\" " +
+		"FOLLOWERS=2 " +
+		"FOLLOWER_CONSTRAINTS=\"[+zone=cn-east-1]\" " +
+		"CONSTRAINTS=\"[+disk=ssd]\"")
+
+	db := testGetSchemaByName(c, tk.Se, "placementDb1")
+	c.Assert(db, NotNil)
+	c.Assert(db.PlacementPolicyRef, IsNil)
+	c.Assert(db.DirectPlacementOpts, NotNil)
+
+	checkFunc := func(policySetting *model.PlacementSettings) {
```

Comment:
It is possible to `c.Assert(policySetting, Equals, &model.PlacementSettings{...})` instead of multi line assertions. But it is just a preference, not a code change request.